### DISCUSSION TITLE
How to save larger than 2GB models?

### DISCUSSION THREAD

**Discussion by rezaalavi** [microsoft/onnxruntime#13989] [Other Q&A]
I am using the following code to convert a model to float16

``` python
import onnxmltools

import onnx
input_onnx_model = '../model/model.onnx'
output_onnx_model = '../onnx/model_fp16.onnx'



#Convert to ONNX ModelProto object and save model binary file:
from onnxmltools.utils.float16_converter import convert_float_to_float16_model_path
new_onnx_model = convert_float_to_float16_model_path(input_onnx_model)
onnx.save_model(new_onnx_model, output_onnx_model)
```

but when it gets to the last line and wants to save the model it throws the following error:
ValueError: Message onnx.ModelProto exceeds maximum protobuf size of 2GB: 12220316957

How do I get to save models that are larger than 2GB?

**Comment by adrianlizarraga** [comment]
Hi @rezaalavi 

Apologies for the delayed response. I would recommend asking this question in the [ONNX github discussions page](https://github.com/onnx/onnx/discussions). You'll likely get a much better answer since that's where the ONNX package is maintained.

That said, you may try loading and saving the model using external data: https://github.com/onnx/onnx/blob/main/docs/ExternalData.md#converting-and-saving-an-onnx-model-to-external-data
